pkg/api/persistentvolumeclaim/util_test.go

```diff
@@ -74,52 +74,38 @@ func TestDropDisabledSnapshotDataSource(t *testing.T) {
 
-	for _, enabled := range []bool{true, false} {
-		for _, oldpvcInfo := range pvcInfo {
-			for _, newpvcInfo := range pvcInfo {
-				oldPvcHasDataSource, oldpvc := oldpvcInfo.hasDataSource, oldpvcInfo.pvc()
-				newPvcHasDataSource, newpvc := newpvcInfo.hasDataSource, newpvcInfo.pvc()
-				if newpvc == nil {
-					continue
-				}
+	for _, oldpvcInfo := range pvcInfo {
+		for _, newpvcInfo := range pvcInfo {
+			oldpvc := oldpvcInfo.pvc()
+			newPvcHasDataSource, newpvc := newpvcInfo.hasDataSource, newpvcInfo.pvc()
+			if newpvc == nil {
+				continue
+			}
 
-				t.Run(fmt.Sprintf("feature enabled=%v, old pvc %v, new pvc %v", enabled, oldpvcInfo.description, newpvcInfo.description), func(t *testing.T) {
-					defer featuregatetesting.SetFeatureGateDuringTest(t, utilfeature.DefaultFeatureGate, features.VolumeSnapshotDataSource, enabled)()
+			t.Run(fmt.Sprintf("old pvc %v, new pvc %v", oldpvcInfo.description, newpvcInfo.description), func(t *testing.T) {
+				var oldpvcSpec *core.PersistentVolumeClaimSpec
+				if oldpvc != nil {
+					oldpvcSpec = &oldpvc.Spec
+				}
+				DropDisabledFields(&newpvc.Spec, oldpvcSpec)
 
-					var oldpvcSpec *core.PersistentVolumeClaimSpec
-					if oldpvc != nil {
-						oldpvcSpec = &oldpvc.Spec
-					}
-					DropDisabledFields(&newpvc.Spec, oldpvcSpec)
+				// old pvc should never be changed
+				if !reflect.DeepEqual(oldpvc, oldpvcInfo.pvc()) {
+					t.Errorf("old pvc changed: %v", diff.ObjectReflectDiff(oldpvc, oldpvcInfo.pvc()))
+				}
 
-					// old pvc should never be changed
-					if !reflect.DeepEqual(oldpvc, oldpvcInfo.pvc()) {
-						t.Errorf("old pvc changed: %v", diff.ObjectReflectDiff(oldpvc, oldpvcInfo.pvc()))
-					}
+				// new pvc should not be changed
+				if !reflect.DeepEqual(newpvc, newpvcInfo.pvc()) {
+					t.Errorf("new pvc changed: %v", diff.ObjectReflectDiff(newpvc, newpvcInfo.pvc()))
+				}
 
-					switch {
-					case enabled || oldPvcHasDataSource:
-						// new pvc should not be changed if the feature is enabled, or if the old pvc had DataSource
-						if !reflect.DeepEqual(newpvc, newpvcInfo.pvc()) {
-							t.Errorf("new pvc changed: %v", diff.ObjectReflectDiff(newpvc, newpvcInfo.pvc()))
-						}
-					case newPvcHasDataSource:
-						// new pvc should be changed
-						if reflect.DeepEqual(newpvc, newpvcInfo.pvc()) {
-							t.Errorf("new pvc was not changed")
-						}
-						// new pvc should not have DataSource
-						if !reflect.DeepEqual(newpvc, pvcWithoutDataSource()) {
-							t.Errorf("new pvc had DataSource: %v", diff.ObjectReflectDiff(newpvc, pvcWithoutDataSource()))
-						}
-					default:
-						// new pvc should not need to be changed
-						if !reflect.DeepEqual(newpvc, newpvcInfo.pvc()) {
-							t.Errorf("new pvc changed: %v", diff.ObjectReflectDiff(newpvc, newpvcInfo.pvc()))
-						}
+				// new pvc should have DataSource
+				if newPvcHasDataSource {
+					if reflect.DeepEqual(newpvc, pvcWithoutDataSource()) {
```

Comment:
Do we need this check? It seems like comparing that new pvc is unchanged above should capture this case already.

Reply:
Sure.  Will remove this.